packages/manager/core/application/src/ovh-tracking.tsx

```diff
@@ -0,0 +1,131 @@
+  'osds-select',
+  'osds-radio-group',
+  'osds-checkbox',
+];
+
+export default function OvhTracking() {
+  const location = useLocation();
+  const shell = useShell();
+  const { tracking, environment } = shell;
+  const env = environment.getEnvironment();
+  const [locationPath, setLocationPath] = useState(location);
+  const myStateRef = useRef(locationPath);
+
+  // Use trackingPage to send action load
+  const OvhTrackPage = (loc: any) => {
+    const path = loc?.pathname.split('/')[1];
+    const partsReplace = loc.pathname.substring(1).replaceAll('/', '::');
+    env.then((response) => {
+      const { applicationName, universe } = response;
+      const name = `${universe}::${applicationName}::${partsReplace ||
+        'homepage'}`;
+      tracking.trackPage({
+        name,
+        level2: '81',
+        page_category: path || 'homepage',
+      });
+    });
+  };
+
+  useEffect(() => {
+    tracking.init(false);
+  }, []);
+
+  useEffect(() => {
+    // When location change, send tracking load page
+    OvhTrackPage(location);
+    myStateRef.current = location;
+    setLocationPath(location);
+  }, [location]);
+
+  const ovhTrackingSendCLick = (value: string) => {
+    const path = myStateRef.current.pathname.split('/')[1];
+    env.then((response) => {
+      const { applicationName, universe } = response;
+      const name = `${applicationName}::${path || 'homepage'}::${value}`;
+      const page = `${universe}::${applicationName}::${path || 'homepage'}`;
+      tracking.trackClick({
+        name,
+        page: { name: page },
+        complete_page_name: page,
+        level2: '81',
+        type: 'action',
+      });
+    });
+  };
+
+  // Send tracking click for the ODS_COMPONENTS_GROUP
+  const ovhTrackingParenNode = (event) => {
+    const value = event.detail.value || event.detail.newValue;
+    ovhTrackingSendCLick(value);
+  };
+
+  // Init event listener for each ODS_COMPONENTS_GROUP on value change
+  const addOdsEventListenerValue = () => {
+    const elmTab = [];
+    ODS_COMPONENTS_GROUP.forEach((element) => {
+      const odsSelector = document.querySelectorAll(element);
+      odsSelector.forEach((elementSlctr) => {
+        elementSlctr?.addEventListener(
+          'odsValueChange',
+          ovhTrackingParenNode,
+          false,
+        );
+        elmTab.push(elementSlctr);
+      });
+    });
+    return elmTab;
+  };
+
+  useLayoutEffect(() => {
+    let tabsEvents = [];
+    const timer = setTimeout(() => {
+      // After the layout and all ods components loaded
+      // Add events listener
+      tabsEvents = addOdsEventListenerValue();
+    }, 500);
+
+    // Clean every listener events
+    return () => {
+      clearTimeout(timer);
+      if (tabsEvents.length > 0)
+        tabsEvents.forEach((element) => {
+          element.removeEventListener(
+            'odsValueChange',
+            ovhTrackingParenNode,
+            false,
+          );
+        });
+    };
+  }, [location]);
```

Comment:
**question:**  Default tracking is added to ODS components when the location changes. But what if some ODS components are added to the DOM after some user actions on the current screen.

For ex,
1. User lands on the config page
2. Enters the value in `Field A`
3. Based on the response to `Field A`, we decide to display either `Field B` or `Field C`.

So with this implementation, are the fields displayed conditionally are not tracked right as there is no change in the location?